### PR TITLE
Notify Slack and email of all claims eligible to send to MAS

### PR DESCRIPTION
**Pull request by nanotone**
## Description of change
This PR expands the eligibility criteria for a Form526 claim establishment to be forwarded to MAS. MAS-eligible claims now include all claims except those that have a pending EP, a corresponding NOTSVCCON disability rating, or are a single-issue HTN claim for increase -- see issue(s) below for context behind these criteria.

Therefore, the Form526RapidReadyForDecisionConcern must be more flexible in supporting different logic for the narrow MAS-specific set of claims, and the broader MAS-eligible set. (For example, the broader set is only being sent to Slack/email for now, with MAS APCAS handoff coming later). To that end, I've reorganized the code a bit:
- Add corresponding `read_metadata` util to mirror `save_metadata` -- streamlines storing form-specific flags to modulate behavior at various times in the form's lifecycle. Suggested in a previous [PR comment](https://github.com/department-of-veterans-affairs/vets-api/pull/10851#discussion_r989423789).
- Made a bunch of methods private. Suggested in a previous [PR comment](https://github.com/department-of-veterans-affairs/vets-api/pull/10851#pullrequestreview-1133554034)
- New small methods `#single_issue?`, `#single_issue_hypertension_cfi?`, and `#increase_only?` to make it easier and more readable to flexibly compose criteria out of these conditions.

## Original issue(s)
department-of-veterans-affairs/abd-vro#491
Immediate upcoming work: department-of-veterans-affairs/abd-vro#492

## Things to know about this PR
* New config variable `Settings.rrd.mas_all_claims_tracking.recipients` with the same list of recipients in all environments, for now.
  * Devops PR: department-of-veterans-affairs/devops#12089
  * k8s ConfigMap PR to come.
* New feature flag `rrd_mas_all_claims_tracking`
* Any PII concerns -- notifications of claim establishments are newly being exported to Slack and email recipients, but the format is identical to, and uses the same code as, the existing `RrdMasNotificationMailer` which exposes no PII.

Testing
* New and updated rspec tests
* Feature flag to be turned on in staging before promoting to prod

### CODE REVIEW

**Review by holdenhinkle — APPROVED** [review]

File: spec/jobs/evss/disability_compensation_form/submit_form526_spec.rb [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
     context 'when the base class is used' do
       it 'raises an error as a subclass should be used to perform the job' do
+        allow_any_instance_of(Form526Submission).to receive(:prepare_for_evss!).and_return(nil)
```
Comment (yoomlam) [inline spec/jobs/evss/disability_compensation_form/submit_form526_spec.rb:29]: Why was this added?

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb [hunk 2]
```diff
@@ -94,27 +78,44 @@ def diagnostic_codes
...
+    if Flipper.enabled?(:rrd_mas_disability_tracking) && single_disability_eligible_for_mas?
+      save_metadata(forward_to_mas: true)
+      insert_classification_codes
+    end
   end
 
   def send_post_evss_notifications!
     send_completed_notification if rrd_job_selector.rrd_applicable?
-    notify_mas if form.dig('rrd_metadata', 'forward_to_mas')
+    notify_mas
```
Comment (yoomlam) [inline app/models/concerns/form526_rapid_ready_for_decision_concern.rb:94]: I suggest renaming this to `conditionally_notify_mas`

**Review by yoomlam — APPROVED** [review]
Left non-blocking suggestion

File: spec/jobs/evss/disability_compensation_form/submit_form526_spec.rb (hunk 1 above)
Comment (nanotone) [reply spec/jobs/evss/disability_compensation_form/submit_form526_spec.rb:29]: This (and wherever else I've added this stub) is an integration test for EVSS submission which cares only about the EVSS segment, but includes our RRD behavior in the test surface area. Since `prepare_for_evss!` is only getting more complicated, it's better for this test to just stub it out rather than do all the requisite VCRing.

It's not the best setup but refactoring these tests should be done in a separate PR.

File: app/models/concerns/form526_rapid_ready_for_decision_concern.rb (hunk 2 above)
Comment (nanotone) [reply app/models/concerns/form526_rapid_ready_for_decision_concern.rb:94]: Great suggestion. In the interest of expediency, I'll leave this for the next PR (which will likely land this week).
